### ISSUE TITLE
[Contextual layers] finish attaching contextual layer styles to components

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon-design-kit#758]
## Acceptance criteria

- Finish attaching the contextual layer styles to components that should use them. 

---

**Components**
- [x] Code snippet
- [x] Contained list
- [x] Data table
- [x] Date picker
- [x] Dropdown / Combobox / Multiselect
- [x] File uploader
- [x] Menu
- [x] Menu button
- [x] Pagination
- [x] Popover
- [x] Structured list
- [x] Tabs (contained)
- [x] Tile
- [x] Treeview

**Review**

- [x] Carbon review
- [x] Figma review

**Comment by kingtraceyj** [comment]
Variable styles have been added in a[ branch](https://www.figma.com/file/YAnB1jKx0yCUL29j6uSLpg/branch/VcSRRPYYfASa6fXdPlUhoJ/(v11)-All-themes---Carbon-Design-System?type=design&node-id=453%3A14562&mode=design&t=A5P2SzQ04I6wcUIh-1)

**Open questions**
-Does using contextual layers in Figma make sense for all components? ie. code snippet?
-Should we have a hybrid approach? Some components using a contextual layer token while other using the standard layering model (-01, -02, -03)
-Is there another solution? (would it cause issues if we added a background token to the main figma file?)

**Comment by kingtraceyj** [comment]
We are going to need to do more research on the best way to implement contextual tokens. the scope is bigger than first anticipated. so far, it's a little buggy and not production ready.

**Comment by sstrubberg** [comment]
We need @juanencalada's input here.

**Comment by sstrubberg** [comment]
The first piece of work around this will be some sort of research on how to go about doing this effectively.

**Comment by thyhmdo** [comment]
Had a call with Juan. It'll be a little bit tricky when it comes to nested layers. 

There is one action item that we can start testing out:
- Add local variables to the main token variables: Layers, Border, and Field under a variable theme as Background or Surface. 
- Test the components above

<img width="780" alt="Image" src="https://github.com/user-attachments/assets/57e475e5-c325-473a-a932-ba0cda11ce1c" />

**Comment by ghost** [comment]
Is there any update regarding this? It's a major inconvenience to not have contextual layers in e.g. the dropdown component. Pretty much whenever the dropdown is used, it has to be manually adjusted by either changing the theme of the component or by overwriting the layer token to be contextual.
If there is no imminent fix, could there be a temporary fix?

Related issues I've discovered:
https://github.com/carbon-design-system/carbon-design-kit/issues/887
(Figma): https://www.figma.com/design/YAnB1jKx0yCUL29j6uSLpg?node-id=14032-308516#1101546832

**Comment by thyhmdo** [comment]
@jonathan-nagel-ibm Yes, we noticed the issue with the dropdown component because the menu has the `layer` contextual token, so it matches with the logic of the layer background. We'll still look into this and will have an update soon for dropdown.

**Comment by thyhmdo** [comment]
Updated in variables. See more info in the Read me page.
